src/mongocxx/test/spec/operation.hh

```diff
@@ -39,23 +39,29 @@ bsoncxx::stdx::optional<read_preference> lookup_read_preference(document::view d
 
 class operation_runner {
    public:
-    explicit operation_runner(collection* coll);
-    operation_runner(database* db,
-                     collection* coll,
-                     client_session* session0 = nullptr,
-                     client_session* session1 = nullptr,
-                     client* client = nullptr);
+    explicit operation_runner() noexcept = default;
+
     document::value run(document::view operation);
 
+    operation_runner& set_collection(std::shared_ptr<collection> coll);
```

Comment:
Nice, making these chainable makes it simpler for callers to use.

Reply:
Agreed, and having these as setters rather than as construction parameters seems a lot nicer and more flexible.